src/dynamics/time_integration.jl

```diff
@@ -285,6 +285,8 @@ function timestep!(
     # PARTICLE ADVECTION (always skip 1st step of first_timesteps!)
     not_first_timestep = lf2 == 2
     not_first_timestep && particle_advection!(progn, diagn, model.particle_advection)
+
+    return nothing 
```

Comment:
Is that what Cthulu.jl pointed out that the return value here isn't type stable? It's good!

Reply:
Yeah, Cthulhu flagged this as "yellow", I think type instabilities like this aren't always bad, and the compiler can often figure them out though. 

In my initial attempt I had some other instabilities from the scratch memory that it help me find.